aws_signing_helper/signer.go

```diff
@@ -205,35 +197,22 @@ func GetSigner(opts *CredentialsOpts) (signer Signer, signatureAlgorithm string,
 					" within the PKCS#12 file")
 			}
 			// Not a PEM certificate? Try PKCS#12
-			certificateChain, privateKey, err = ReadPKCS12Data(opts.CertificateId)
+			certificateChain, _, err = ReadPKCS12Data(opts.CertificateId)
 			if err != nil {
 				return nil, "", err
 			}
-			if privateKey != nil {
-				ecPrivateKeyPtr, isEcKey := privateKey.(*ecdsa.PrivateKey)
-				if isEcKey {
-					privateKey = *ecPrivateKeyPtr
-				}
-
-				rsaPrivateKeyPtr, isRsaKey := privateKey.(*rsa.PrivateKey)
-				if isRsaKey {
-					privateKey = *rsaPrivateKeyPtr
-				}
-			}
-			return GetFileSystemSigner(privateKey, certificateChain[0], certificateChain)
+			return GetFileSystemSigner(opts.PrivateKeyId, opts.CertificateId, opts.CertificateBundleId, true)
 		} else {
 			return nil, "", err
 		}
 	}
 
 	if opts.CertificateBundleId != "" {
-		certificateChainPointers, err := ReadCertificateBundleData(opts.CertificateBundleId)
+		chain, err := GetCertChain(opts.CertificateBundleId)
 		if err != nil {
 			return nil, "", err
 		}
-		for _, certificate := range certificateChainPointers {
-			certificateChain = append(certificateChain, certificate)
-		}
+		certificateChain = chain
```

Comment:
Why assign to `certificateChain` after checking the error if we will return from the function anyways in the case that there's an error? `certificateChain` is local anyways, so I don't think there's any risk of assigning `nil` to it right before returning from the function (in the case of error).

Reply:
`chain` is local to the if block and therefore not available when it needs to be returned.  `certificateChain` is scoped to the function and is available on line when this return is called:
`return GetPKCS11Signer(opts.LibPkcs11, certificate, certificateChain, opts.PrivateKeyId, opts.CertificateId, opts.ReusePin)`